notatrix/tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+        "allowJs": true,
+        "outDir": "./build/js",
+        "target": "es5"
```

Comment:
Which "flavor" of JavaScript should we emit?  This is a shockingly confusing corner of the language, but the basic idea is that the `es5` target is basically universally supported at this point.  I think, honestly, a good direction ultimately would be to change this to `es6`, migrate `client/` + `server/` to TypeScript also, and then get rid of Babel.  I really wish the JavaScript ecosystem were a little more stable, as this is all extremely subtle :cry:

Reply:
What do we need that's part of `es6`?

Reply:
IIRC, there's some weirdness in ES5 around extending certain builtin classes like `Array` or `Error`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#es5_customerror_object